base/sort.jl

```diff
@@ -430,7 +430,41 @@ function sort!(v::AbstractVector;
                by=identity,
                rev::Bool=false,
                order::Ordering=Forward)
-    sort!(v, alg, ord(lt,by,rev,order))
+    ordr = ord(lt,by,rev,order)
+    if ordr === Forward && isa(v,Vector) && eltype(v)<:Integer
+        n = length(v)
+        if n > 1
+            min, max = extrema(v)
+            rangelen = max - min + 1
```

Comment:
Thanks. Put up a PR with a possible fix. We should add functions that expose the overflow flag instead of throwing exceptions; LLVM has those but as far as I remember we don't provide direct access to them yet.